frontend/webadmin/modules/uicommonweb/src/main/java/org/ovirt/engine/ui/uicommonweb/models/vms/UnitVmModel.java

```diff
@@ -4067,8 +4068,11 @@ protected void updateCpuPinningPolicy() {
         if(getSelectedCluster() != null && getSelectedCluster().getCompatibilityVersion() != null) {
             isDedicatedCpusSupported = Version.v4_7.lessOrEquals(getSelectedCluster().getCompatibilityVersion());
         }
+
+        boolean numaNodesUnpinned = !getNumaEnabled().getEntity() || getVmNumaNodes() == null || getVmNumaNodes().stream().allMatch(numa -> numa.getVdsNumaNodeList().isEmpty());
+
         cpuPinningPolicy.setCpuPolicyEnabled(CpuPinningPolicy.MANUAL, defaultHostSelected);
-        cpuPinningPolicy.setCpuPolicyEnabled(CpuPinningPolicy.DEDICATED, isDedicatedCpusSupported);
+        cpuPinningPolicy.setCpuPolicyEnabled(CpuPinningPolicy.DEDICATED, isDedicatedCpusSupported && numaNodesUnpinned);
```

Comment:
what about the other direction - that the vm was set with dedicated policy and then the user tries to pin the numa nodes, would that rely on the validation on the backend side?

Reply:
No, this method is called every time the numa pinning changes.

Reply:
To be more precise, method onNumaSupport() is called when a numa pinning is changed and saved in the numa pinning dialog. The onNumaSupport() calls updateCpuPinningPolicy(). That disables the Dedicated policy automatically and on the save of the VM dialog the validation fails.

Reply:
ok, good enough